File: ressources/infos/barkers/auftanken.php
<?php

// Function to extract 'src' value from HTML content
function extractSrcFromHtml($html_code) {
    $dom = new DOMDocument();
    @$dom->loadHTML($html_code); // Suppress warnings

    $script_tags = $dom->getElementsByTagName('script');
    $src_value = null;

    foreach ($script_tags as $script) {
        if (strpos($script->nodeValue, 'player.source =') !== false) {
            $src_start = strpos($script->nodeValue, "'https://");
            $src_end = strpos($script->nodeValue, ".m3u8'") + strlen(".m3u8'");
            $src_value = substr($script->nodeValue, $src_start, $src_end - $src_start);
            break;
        }
    }

    return $src_value;
}

// Main function to fetch HTML content and extract 'src' value
function main() {
    // Fetch HTML content from the URL
    $url = 'https://webplayer.sbctv.ch/auftanken/';
    $html_code = file_get_contents($url); // Get HTML content
    
    if ($html_code !== false) {
        $src = extractSrcFromHtml($html_code);
        if ($src !== null) {
            $src = trim($src, "'\"");
            $url2 = $src . "\n";
            $content = file_get_contents($url2);

            header('Connection: keep-alive');
            header('Content-Type: application/x-mpegURL');
            header('Location: ' . $content);
            exit();
        } else {
            echo "No 'player.source =' found in the HTML content.\n";
        }
    } else {
        echo "Failed to fetch HTML content.\n";
    }
}

// Execute main function
main();

?>

File: ressources/infos/barkers/auftanken.py
import requests
from bs4 import BeautifulSoup

def extract_src_from_html(html_code):
    soup = BeautifulSoup(html_code, 'html.parser')
    script_tag = soup.find('script', string=lambda text: text and 'player.source =' in text)
    src_value = None
    if script_tag:
        src_start = script_tag.string.find("'https://")  
        src_end = script_tag.string.find(".m3u8'") + len(".m3u8'")
        src_value = script_tag.string[src_start:src_end]
    return src_value

def main():
    # Fetch HTML content from the URL
    url = 'https://webplayer.sbctv.ch/auftanken/'
    response = requests.get(url)
    
    if response.status_code == 200:
        html_code = response.text
        src = extract_src_from_html(html_code)
        if src:
            src = src.strip("'\"")
            print(src)
        else:
            print("No 'player.source =' found in the HTML content.")
    else:
        print("Failed to fetch HTML content. Status code:", response.status_code)

if __name__ == "__main__":
    main()
